share/gpodder/extensions/subscription_stats.py

```diff
@@ -0,0 +1,186 @@
+        for channel_id, channel_last, channel_name, paused in channels:
+            if paused:
+                nr_paused += 1
+
+            total = 0
+            nr_episodes = 0
+            prev = now
+            with db.lock:
+                cur = db.cursor()
+                cur.execute('SELECT published FROM %s WHERE podcast_id = %d ORDER BY published DESC LIMIT 25'
+                    % (db.TABLE_EPISODE, channel_id))
+                while True:
+                    row = cur.fetchone()
+                    if row is None:
+                        break
+                    if total == 0:
+                        edate = row[0]
+                    total += (prev - row[0])
+                    nr_episodes += 1
+                    prev = row[0]
+                cur.close()
+
+            average = (total / nr_episodes) / 86400 if nr_episodes > 0 else 0
+            last = strftime('%Y/%m/%d', localtime(edate))
+            if channel_last: 
+                last_feed = datetime.strptime(channel_last, '%a, %d %b %Y %H:%M:%S %Z').strftime('%Y/%m/%d')
+            else:
+                last_feed = ""
+
+            if average == 0:
+                yearly.append([average, channel_name, last, last_feed, paused])
+            elif average <= 2:
+                daily.append([average, channel_name, last, last_feed, paused])
+            elif average <= 14:
+                weekly.append([average, channel_name, last, last_feed, paused])
+            elif average <= 61:
+                monthly.append([average, channel_name, last, last_feed, paused])
+            else:
+                yearly.append([average, channel_name, last, last_feed, paused])
+
+        # sort by averages
+        daily.sort(key=lambda e: e[0])
+        weekly.sort(key=lambda e: e[0])
+        monthly.sort(key=lambda e: e[0])
+        yearly.sort(key=lambda e: e[0])
+
+        # open dialog
+        dlg = Gtk.Dialog(_('Subscription Statistics'), self.gpodder.main_window)
+        dlg.set_size_request(400, 400)
+        dlg.set_resizable(True)
+
+        box = Gtk.Box(orientation=Gtk.Orientation.VERTICAL, spacing=10)
+        box.set_border_width(10)
+
+        label = Gtk.Label(_('%s subscriptions (%d paused)') % (len(channels), nr_paused))
+        box.add(label)
+
+        notebook = Gtk.Notebook()
+        notebook.set_vexpand(True)
+        self.add_page(notebook, _('daily'), daily)
+        self.add_page(notebook, _('weekly'), weekly)
+        self.add_page(notebook, _('monthly'), monthly)
+        self.add_page(notebook, _('yearly'), yearly)
+        box.add(notebook)
+
+        label = Gtk.Label(_('Average days between the last 25 episodes.'))
+        label.set_line_wrap(True)
+        box.add(label)
+
+        hbox = Gtk.Box(orientation=Gtk.Orientation.HORIZONTAL, spacing=0)
+        button = Gtk.Button(stock=Gtk.STOCK_CLOSE)
+        button.connect('clicked', lambda w: dlg.destroy())
+        hbox.pack_end(button, False, False, 0)
+        box.add(hbox)
```

Comment:
It would be easier to just use `dlg.add_button(_('_Close'), Gtk.ResponseType.OK)` here. That way you also don't need the deprecated GtkStock.

Reply:
Ok.